sale_blanket_order/models/blanket_orders.py

```diff
@@ -0,0 +1,331 @@
+        default.update(self.default_get(['name', 'confirmed']))
+        return super(BlanketOrder, self).copy_data(default)
+
+    @api.multi
+    def _validate(self):
+        try:
+            today = fields.Date.today()
+            for order in self:
+                assert order.validity_date, _("Validity date is mandatory")
+                assert order.validity_date > today, \
+                    _("Validity date must be in the future")
+                assert order.partner_id, _("Partner is mandatory")
+                assert len(order.lines_ids) > 0, _("Must have some lines")
+                order.lines_ids._validate()
+        except AssertionError as e:
+            raise UserError(e.message)
+
+    @api.multi
+    def action_confirm(self):
+        self._validate()
+        for order in self:
+            sequence_obj = self.env['ir.sequence']
+            if order.company_id:
+                sequence_obj = sequence_obj.with_context(
+                    force_company=order.company_id.id)
+            name = sequence_obj.next_by_code('sale.blanket.order')
+            order.write({'confirmed': True, 'name': name})
+        return True
+
+    @api.multi
+    def action_view_sale_orders(self):
+        sale_orders = self._get_sale_orders()
+        action = self.env.ref('sale.action_orders').read()[0]
+        if len(sale_orders) > 0:
+            action['domain'] = [('id', 'in', sale_orders.ids)]
+        else:
+            action = {'type': 'ir.actions.act_window_close'}
+        return action
+
+    @api.model
+    def expire_orders(self):
+        today = fields.Date.today()
+        expired_orders = self.search([
+            ('state', '=', 'opened'),
+            ('validity_date', '<=', today),
+        ])
+        expired_orders.modified(['validity_date'])
+        expired_orders.recompute()
+
+
+class BlanketOrderLine(models.Model):
+    _name = 'sale.blanket.order.line'
+    _description = 'Blanket Order Line'
+
+    sequence = fields.Integer()
+    order_id = fields.Many2one(
+        'sale.blanket.order', required=True, ondelete='cascade')
+    product_id = fields.Many2one(
+        'product.product', string='Product', required=True)
+    product_uom = fields.Many2one(
+        'product.uom', string='Unit of Measure', required=True)
+    price_unit = fields.Float(string='Price', required=True)
+    original_qty = fields.Float(
+        string='Original quantity', required=True, default=1,
+        digits=dp.get_precision('Product Unit of Measure'))
+    ordered_qty = fields.Float(
+        string='Ordered quantity', compute='_compute_quantities', store=True)
+    invoiced_qty = fields.Float(
+        string='Invoiced quantity', compute='_compute_quantities', store=True)
+    remaining_qty = fields.Float(
+        string='Remaining quantity', compute='_compute_quantities', store=True)
+    delivered_qty = fields.Float(
+        string='Delivered quantity', compute='_compute_quantities', store=True)
```

Comment:
I think the store=True parameter should be removed from ordered_qty, invoiced_qty, remaining_qty and delivered_qty to fix that issue.

Reply:
@bjeficent Why removing store=True ?